localstack/services/opensearch/resource_providers/aws_opensearchservice_domain.py

```diff
@@ -200,41 +202,51 @@ def create(
 
         """
         model = request.desired_state
-
-        # Validations
-        assert model["DomainName"]
-
+        if tags := model.pop("Tags", []):
+            model["TagList"] = tags
```

Comment:
```suggestion
        create_kwargs = {
            **util.deselect_attributes(model, ["Tags"]),
            "TagList": model.get('Tags')
        }
```

I'd also suggest moving this farther down to just before the `create_domain` call

In the future we'll be quite strict here and this would fail since it doesn't conform to the resource schema.
Only properties defined in the `OpenSearchServiceDomainProperties` typed dict should be set on the model.
Pycharm should've actually also complained about this :thinking: